### PR TITLE
packageOperations should be a map between operation type and list of …

### PR DESCRIPTION
**Pull request by shainer**
…packages. It does not make sense for it to be a list of dictionaries, all with the same format. Also makes conversions with QVariant harder.

Moreover, add conversion from/to pydict and QHash, mirroring the existing implementation for QMap. Nice addition :)

Found these while testing the netinstall module.

### CODE REVIEW

File: src/libcalamares/PythonHelper.cpp [hunk 1]
```diff
@@ -165,20 +197,16 @@ Helper::Helper( QObject* parent )
...
         if ( fi.exists() && fi.isReadable() )
-        {
             m_pythonPaths.append( fi.dir().absolutePath() );
-        }
 
         QDir calaPythonPath( CalamaresUtils::systemLibDir().absolutePath() +
                              QDir::separator() + "calamares" );
         if ( calaPythonPath.exists() &&
-             calaPythonPath.isReadable() )
+                calaPythonPath.isReadable() )
```
Comment (teo) [inline src/libcalamares/PythonHelper.cpp]: Any reason for this whitespace change?

File: src/libcalamares/PythonHelper.cpp [hunk 2]
```diff
@@ -94,9 +98,7 @@ variantListToPyList( const QVariantList& variantList )
 {
     bp::list pyList;
     foreach ( const QVariant& variant, variantList )
-    {
```
Comment (teo) [inline src/libcalamares/PythonHelper.cpp]: I kinda liked these braces even for 1-line blocks. Keeps everyone from making mistakes if adding lines at some point.

File: src/modules/packages/main.py [hunk 3]
```diff
@@ -124,9 +124,6 @@ def run():
         run_operations(pkgman, entry)
 
     if libcalamares.globalstorage.contains("packageOperations"):
-        operations = libcalamares.globalstorage.value("packageOperations")
-
-        for entry in operations:
-            run_operations(pkgman, entry)
+        run_operations(libcalamares.globalstorage.value("packageOperations"))
```
Comment (teo) [inline src/modules/packages/main.py]: This won't work as long as run_operations requires a package manager as first argument.

File: src/libcalamares/PythonHelper.cpp (hunk 2 above)
Comment (shainer) [inline src/libcalamares/PythonHelper.cpp]: Both this and the whitespace change were done by astyle (the script in hacking/). We should fix that script :)
I'll correct the Python invocation soon.

File: src/modules/packages/main.py (hunk 3 above)
Comment (shainer) [inline src/modules/packages/main.py]: Thanks, done.

**Pull request merged**